app/src/utils/apiOperations/notebooks.tsx

```diff
@@ -29,3 +36,102 @@ export const createNotebookFromTemplate = async (input: {
     {template_id: input.templateId, name: input.name} as PostCreateNotebookInput
   );
 };
+
+export interface RecordStatus extends PostRecordStatusResponse {
+  recordHashes: Record<string, string>;
+}
+
+/**
+ * Validate the sync status of records in a project
+ *
+ * @param projectId - the project id to validate
+ */
+export const validateSyncStatus = async ({
+  projectId,
+  username,
+  listingId,
+  currentStatus,
+}: {
+  projectId: ProjectID;
+  username: string;
+  listingId: string;
+  currentStatus: RecordStatus | undefined;
+}): Promise<RecordStatus> => {
+  // get the list of record ids from the project
+  const dataDb = await getDataDB(projectId);
```

Comment:
getDataDB I am wanting to deprecate - it's the annoying registered client thing. Would prefer 'localGetDataDB' (don't love the name) - that uses the redux store (which is what is registered in this data model anyway).